src/main/java/io/github/thebusybiscuit/slimefun4/implementation/setup/ResearchSetup.java

```diff
@@ -287,6 +287,7 @@ public static void setupResearches() {
         register("improved_generators", 275, "Improved Generators", 24, SlimefunItems.COAL_GENERATOR_2, SlimefunItems.LAVA_GENERATOR_2);
         register("ingredients_and_cheese", 276, "Slimefun Cuisine", 5, SlimefunItems.SALT, SlimefunItems.WHEAT_FLOUR, SlimefunItems.HEAVY_CREAM, SlimefunItems.CHEESE, SlimefunItems.BUTTER);
         register("medium_tier_auto_enchanting", 277, "Fast Automatic Enchanting and Disenchanting", 48, SlimefunItems.AUTO_ENCHANTER_2, SlimefunItems.AUTO_DISENCHANTER_2);
+        register("rainbow_armor", 279, "Wear the Rainbow", 22, SlimefunItems.RAINBOW_HELMET, SlimefunItems.RAINBOW_CHESTPLATE, SlimefunItems.RAINBOW_LEGGINGS, SlimefunItems.RAINBOW_BOOTS);
```

Comment:
> Even if you reserved that id for a different PR

Yes, I did (the portable teleporter which is close to being merged).
How else are we supposed to number these?

Reply:
Next available.
Because if we somehow decided against the merging of the Portable Teleporter PR for example, there would have been a gap.
And when we did merge it, this results in a merge conflict either way. So no matter what, there will be a secondary action required if you skip a number here.
By choosing the next available id at the time, you'd only have to take actions in the event of a successful merge, otherwise not.
This is more efficient.